src/vehicle.cpp

```diff
@@ -2539,6 +2539,8 @@ void Vehicle::LeaveUnbunchingDepot()
 
 		u->depot_unbunching_next_departure = next_departure;
 	}
+
+	SetWindowClassesDirty(WC_VEHICLE_VIEW);
```

Comment:
Is it really necessary that the view of *all* vehicles in the game need to be dirtied? Isn't it only the vehicle that got unbunched, as the state for the others should not have changed?

Note that it's not necessarily bad to do it for a whole class when there can be multiple, and the window is not shown often. As then trying to find many windows might be more expensive than occasionally redrawing the window even when it's not needed. It's just that above you only dirty the timetable of the current vehicle.

Reply:
All vehicles that share orders with this one must be dirtied. I moved the call inside the loop.

We only dirty the timetable of the current vehicles because it's the only one marked "on time" -- other vehicles in the shared-order group are not changed.